tests/ui/asm_syntax.rs

```diff
@@ -0,0 +1,31 @@
+#![feature(asm)]
+// only-x86 only-x86_64
```

Comment:
```suggestion
// only-x86_64
```
It seems that two `only` header commands refering to different targets should not be used in the same file, otherwise the test will never run given that both would have to be true at the same time. See [here](https://github.com/rust-lang/rust/blob/867bd42c38bac3870388765f65386b18cf1d9c5f/src/tools/compiletest/src/header.rs#L61-L65) for details. Maybe that's the reason why rustc tests use only x86_64.

BTW, while checking this I've realized that [compiletest-rs](https://github.com/laumann/compiletest-rs), which is the "extracted" version that Clippy uses (the original lives in-tree in the compiler repo), does not support the `only` directive... I don't think we should use `ignore-X` with each and every unsupported architecture, as it would be a pain to maintain. 

Let's leave it as `// only-x86_64` (I've applied the change) for now as AFAIK our CI does not test architectures where this would fail. I will try to add support for `only` in compiletest-rs.